src/ghga_transpiler/core.py

```diff
@@ -20,11 +20,29 @@
 from typing import Callable, Optional, Union
 
 from openpyxl import Workbook
+from packaging.version import InvalidVersion, Version
 
 from . import config
 
-# pylint: disable=line-too-long
-SEMVER_REGEX = r"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+
+class InvalidSematicVersion(Exception):
+    """Raised when a version string is invalid."""
+
+
+SEMVER_PATTERN = r"^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$"
+
+
+class SemanticVersion(Version):
+    """Extends Version class to check if a version follows the rules of semantic versioning"""
+
+    _semver = re.compile(r"^\s*" + SEMVER_PATTERN + r"\s*$", re.VERBOSE | re.IGNORECASE)
+
+    def __init__(self, wb_version):
+        super().__init__(wb_version)
+
+        semver_match = self._semver.search(wb_version)
+        if not semver_match:
+            raise InvalidVersion(f"Invalid version: '{wb_version}'")
```

Comment:
Maybe using this library instead would save us from changing the other library to behave semver-like?

https://python-semver.readthedocs.io/en/latest/

Reply:
I did a quick search before implementation and found that the internet was quite fond of `packaging`, mainly because it is also used in `setuptools`. But I can switch to python-semver.